src/lib.rs

```diff
@@ -380,6 +381,12 @@ impl WalkDir {
     /// paths in sorted order. The compare function will be called to compare
     /// entries from the same directory.
     ///
+    /// If `sort_by` is not called, the order of paths is unspecified. WalkDir
+    /// calls `std::fs::read_dir` to get the directory contents. Currently on
+    /// Unix this uses `opendir` and `readdir`, and on Windows it uses
+    /// `FindFirstFile` and `FindNextFile`. In both cases the order is
+    /// determined by the underlying filesystem.
```

Comment:
I'm against this documentation, sorry. I don't think we should be putting implementation details like this into the public API.

I am confused as to why the current documentation is insufficient. I'm in favor of repeating the "unspecified order" bit in more places if that would help, but I would rather not give much more detail than that.

Reply:
Rust itself lists it in [its public API docs](https://doc.rust-lang.org/std/fs/fn.read_dir.html#platform-specific-behavior) - I'm just duplicating that.

Basically, for my use case (hashing a load of files) it is significantly slower if I have to sort them first. But also, I don't mind exactly what the order is as long as it only changes when the files on disk change. So there is an intermediate step between "it will be totally random every time", and "it's sorted" which is useful for me, and also is what the actual behaviour is. Does that make sense?

I've actually since changed my actual code to use an order-independent hash, but still I feel like it is helpful to have this information here, and it is surely unlikely to change? (And I did put `Currently`...)

Reply:
> I don't mind exactly what the order is as long as it only changes when the files on disk change

I don't think this property is ever guaranteed, or at least, it's likely dependent on the file system. It's not up to this crate.

But yeah, I know `std`'s docs say this, but that doesn't mean I have to agree with doing that. I really, really do not think the implementation details should be in this doc. Sorry.

And yes, I do absolutely think this could change in the future. I have often mulled over the possibility of skipping `std:fs::read_dir` and implementing the platform specific stuff myself, but I don't know if it's worth it.